specs/incremental.go

```diff
@@ -0,0 +1,49 @@
+package specs
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+)
+
+type Incremental int
+
+const (
+	IncrementalNone Incremental = iota
+	IncrementalTablesOnly
+	IncrementalBoth
```

Comment:
What is the behavior expected to be for all these settings? What is the difference between setting `IncrementalTablesOnly` here and using the `OnlyIncrementalTables` boolean at the source spec level?